wallet/test/Migrations.hs

```diff
@@ -0,0 +1,206 @@
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+{-# LANGUAGE FlexibleInstances          #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE RecordWildCards            #-}
+{-# LANGUAGE StandaloneDeriving         #-}
+{-# LANGUAGE TypeFamilies               #-}
+module Migrations (tests) where
+
+import           Control.Arrow                        ((***))
+import qualified Data.HashMap.Strict                  as HM
+import           Data.SafeCopy
+import           Pos.Arbitrary.Core                   ()
+import           Pos.Wallet.Web.ClientTypes           (AccountId (..), Addr,
+                                                       CAccountMeta (..), CCoin (..),
+                                                       CHash (..), CId (..),
+                                                       CProfile (..), CTxId (..),
+                                                       CTxMeta (..), CUpdateInfo (..),
+                                                       CWAddressMeta (..),
+                                                       CWalletAssurance (..),
+                                                       CWalletMeta (..), Wal)
+import           Pos.Wallet.Web.ClientTypes.Functions (addressToCId)
+import           Pos.Wallet.Web.State.Storage
+import           Test.QuickCheck
+import           Test.Tasty
+import           Test.Tasty.QuickCheck
+import           Universum
+
+instance Migrate AddressInfo_v0 where
```

Comment:
Probably best to call out explicitly what we're doing here, that we're setting up "reverse" migrations, to go from newer to older. This then explains the `migrate . migrate` in the test at the bottom, which is going in two different directions.